File: Python/Programmars/Level_1/내적.py
def solution(a, b):
    answer = 0
    for i in range(len(a)):
        answer += a[i] * b[i]
    return answer


print(solution([1, 2, 3, 4], [-3, -1, 0, 2]))  # 3
print(solution([-1, 0, 1], [1, 0, -1]))  # -2


# ref
def solution(a, b):
    return sum([x*y for x, y in zip(a, b)])

File: JavaScript/Programmars/Level_1/내적.js
function solution(a, b) {
  let answer = 0;
  const len = a.length;
  for (let i = 0; i < len; i++) {
    answer += a[i] * b[i];
  }

  return answer;
}

console.log(solution([1, 2, 3, 4], [-3, -1, 0, 2])); // 3
console.log(solution([-1, 0, 1], [1, 0, -1])); // -2

// 다른 풀이: reduce 활용
function solution(a, b) {
  return a.reduce((acc, _, i) => (acc += a[i] * b[i]), 0);
}
